ocp_utilities/must_gather.py

```diff
@@ -51,3 +54,37 @@ def run_must_gather(
         flag_string = "".join([f" --{flag_name}" for flag_name in flag_names])
         base_command += f" {flag_string}"
     return run_command(command=shlex.split(base_command), check=False)[1]
+
+
+def collect_must_gather(must_gather_output_dir, cluster_name, product_name, kubeconfig_path=None):
+    """
+    Run must-gather command with an option to create target directory using must_gather_output_dir, cluster_name, product_name.
+    (This function can be used for running must-gather incase any product installation fails.)
+
+    Args:
+        must_gather_output_dir (str): Path to base directory where must-gather logs will be stored
+        cluster_name (str): Cluster Name for which must-gather will run
+        product_name (str): Product Name installed on given cluster
+        kubeconfig_path (str, optional): Path to kubeconfig, If not provided KUBECONFIG env variable will be used
+    """
+
+    target_dir = os.path.join(must_gather_output_dir, "must-gather", product_name, cluster_name)
+
+    try:
+        LOGGER.info(f"Prepare must-gather target extracted directory {target_dir}.")
+        Path(target_dir).mkdir(parents=True, exist_ok=True)
+
+        LOGGER.info(f"Collect must-gather for cluster {cluster_name}")
+        run_must_gather(
+            target_base_dir=target_dir,
+            kubeconfig=kubeconfig_path or os.environ.get("KUBECONFIG"),
```

Comment:
this should not be done here.
`kubeconfig_path` - pass it as is; `run_must_gather` takes care of it.